haskell/src/Monocle/Backend/Index.hs

```diff
@@ -415,44 +414,101 @@ getDocument bhEnv index dId = do
+        toString crawlerTypeValue
+      ]
+
+getCrawlerMetadata :: BH.BHEnv -> BH.IndexName -> CrawlerMetadataDocId -> IO (Maybe ELKCrawlerMetadata)
 getCrawlerMetadata = getDocument
 
 getLastUpdatedFromConfig :: UTCTime
 getLastUpdatedFromConfig = parseTimeOrError False defaultTimeLocale "%F" "2021-01-01"
 
 data Entity = Project {getName :: Text} | Organization {getName :: Text}
 
-getEntityName :: Entity -> Text
-getEntityName entity = case entity of
-  Project name -> toText $ intercalate "-" ["project", toString name]
-  Organization name -> toText $ intercalate "-" ["organization", toString name]
-
 type EntityType = CrawlerPB.CommitInfoRequest_EntityType
 
-getLastUpdated :: BH.BHEnv -> BH.IndexName -> Config.Worker -> EntityType -> IO (Text, UTCTime)
-getLastUpdated _bhEnv _index _worker _entity = undefined
+getWorkerName :: Config.TaskCrawler -> Text
+getWorkerName Config.TaskCrawler {..} = name
 
-{-
-  cmM <- getCrawlerMetadata bhEnv index cmID
-  case cmM of
-    Just cm -> pure $ elkcmLastCommitAt cm
-    Nothing -> pure getLastUpdatedFromConfig
+getLastUpdated :: BH.BHEnv -> BH.IndexName -> Config.Worker -> EntityType -> IO (Text, UTCTime)
+getLastUpdated bhEnv index worker entity = do
+  BH.runBH bhEnv $ do
+    let search' = search
+    print search'
+    resp <- fmap BH.hitSource <$> (Q.simpleSearch index search :: BH.BH IO [BH.Hit ELKCrawlerMetadata])
+    case catMaybes resp of
+      [] -> error "Unsupported"
+      (x : _) -> pure $ getRespFromMetadata x
   where
-    cmID = getCrawlerMetadataID entity
--}
+    search =
+      (BH.mkSearch (Just query) Nothing)
+        { BH.size = BH.Size 1,
+          BH.sortBody =
+            Just
+              [ BH.DefaultSortSpec
+                  ( BH.DefaultSort (BH.FieldName "crawler_metadata.last_commit_at") BH.Ascending Nothing Nothing Nothing Nothing
+                  )
+              ]
+        }
+    query =
+      BH.QueryBoolQuery $
+        BH.mkBoolQuery
+          [ BH.TermQuery (BH.Term "crawler_metadata.crawler_name" (getWorkerName worker)) Nothing,
+            BH.TermQuery (BH.Term "crawler_metadata.crawler_type" (crawlerType entity)) Nothing
+          ]
+          []
+          []
+          []
+    crawlerType :: EntityType -> Text
+    crawlerType entity' = case entity' of
+      CrawlerPB.CommitInfoRequest_EntityTypeProject -> "project"
+      _ -> error "Unsupported Entity"
+    getRespFromMetadata (ELKCrawlerMetadata ELKCrawlerMetadataObject {..}) = (toStrict elkcmCrawlerTypeValue, elkcmLastCommitAt)
 
-setLastUpdated :: BH.BHEnv -> BH.IndexName -> Entity -> UTCTime -> IO ()
-setLastUpdated bhEnv index entity lastUpdatedDate = do
+setLastUpdated :: BH.BHEnv -> BH.IndexName -> Text -> UTCTime -> Bool -> Entity -> IO ()
+setLastUpdated bhEnv index crawlerName lastUpdatedDate doNotUpdate entity = do
   BH.runBH bhEnv $ do
     exists <- BH.documentExists index id'
     resp <-
       if exists
         then do
+          when doNotUpdate $ pure ()
```

Comment:
The IO Monad does not stop on unit, thus this always update. You need to use `unless` here:

```suggestion
          unless doNotUpdate $ BH.updateDocument index BH.defaultIndexDocumentSettings cm id'
```